### PR TITLE
Infinite scroll

### PR DESCRIPTION
**Pull request by fvictorio**
This has several pending issues:

- The infinite scroll it's not a scroll. Instead, a "Show more" button is used.
- List behaves wrongly after the user connects his wallet.

### CODE REVIEW

File: app/src/components/market/market_home.tsx [hunk 1]
```diff
@@ -38,11 +42,16 @@ export const MarketHome: React.FC<Props> = (props: Props) => {
         />
       )}
       <ListCard>
-        {RemoteData.is.success(markets)
-          ? markets.data.map((item, index) => {
-              return <ListItem key={index} data={item}></ListItem>
+        {RemoteData.hasData(markets)
+          ? markets.data.slice(0, props.count).map(item => {
```
Comment (mariano-aguero) [inline app/src/components/market/market_home.tsx]: I think we can take advantage of the destructuring in the line 31, and add `count` and `onShowMore` to the destructuring

File: app/src/services/market_maker_factory.ts [hunk 2]
```diff
@@ -51,13 +60,14 @@ class MarketMakerFactoryService {
     return marketMakerAddress
   }
 
-  getMarkets = async (provider: any): Promise<Market[]> => {
+  getMarkets = async ({ from, to }: GetMarketsOptions): Promise<Market[]> => {
+    console.log(`fetching from ${from} to ${to}`)
```
Comment (mariano-aguero) [inline app/src/services/market_maker_factory.ts]: Should be `logger.log`

File: app/src/hooks/useMarkets.tsx [hunk 3]
```diff
@@ -1,113 +1,154 @@
...
-import { MarketMakerService } from '../services/market_maker'
-import { getLogger } from '../util/logger'
-import { Market, MarketWithExtraData, MarketFilters, MarketStatus } from '../util/types'
+import { useContracts, Contracts } from './useContracts'
+import { MarketWithExtraData, MarketFilters } from '../util/types'
+import { callInChunks, Range } from '../util/call_in_chunks'
 import { RemoteData } from '../util/remote_data'
 
-const logger = getLogger('Market::useMarkets')
+const EARLIEST_BLOCK_TO_CHECK = 4986777
```
Comment (mariano-aguero) [inline app/src/hooks/useMarkets.tsx]: This should be an env var , we can added this value `4986777` as a default value if the env var doesn't exist.

File: app/src/components/market/market_home.tsx (hunk 1 above)
Comment (fvictorio) [reply app/src/components/market/market_home.tsx]: Just for the record, I kind of like to use `props.foo` directoy when `foo` is only used once. That being said, I changed it for consistency with what we're already doing.

File: app/src/hooks/useMarkets.tsx (hunk 3 above)
Comment (fvictorio) [reply app/src/hooks/useMarkets.tsx]: Done. Also added the chunk size as an env var.

File: app/src/services/market_maker_factory.ts (hunk 2 above)
Comment (fvictorio) [reply app/src/services/market_maker_factory.ts]: Done.

Comment (fvictorio) [reply app/src/services/market_maker_factory.ts]: Also added the `no-console` rule to our eslint config btw.

**Pull request merged**